clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/rocksdb/ReplicationMetadataRocksDBStoragePartition.java

```diff
@@ -206,6 +206,7 @@ public synchronized void endBatchWrite() {
 
     if (deferredWrite) {
       rocksDBSstFileWriter.ingestSSTFiles(rocksDB, getColumnFamilyHandleList());
+      rocksDBSstFileWriter.createSnapshot(rocksDB);
```

Comment:
We need to control this new behavior, including some codes below, via a feature flag so we won't create snapshot for every stores.. I think we can use `StoragePartitionConfig` or `VeniceStoreVersionConfig` for this. Can you take a look and add that?

Reply:
Also, you probably don't need it here since `ReplicationMetadataRocksDBStoragePartition` extends `RocksDBStoragePartition` and you have added the behavior in `RocksDBStoragePartition#endBatchWrite`

Reply:
I am not sure I follow. When `deferredWrite` is false, the method returns for both `ReplicationMetadataRocksDBStoragePartition` and `RocksDBStoragePartition`. When it is true, we end up calling `createSnapshot` twice.